tests/datasets/test_streaming_datasets_train.py

```diff
@@ -0,0 +1,83 @@
+# Copyright 2022 MosaicML Composer authors
+# SPDX-License-Identifier: Apache-2.0
+
+import os
+from pathlib import Path
+
+import numpy as np
+import pytest
+from torch.utils.data import DataLoader
+
+from composer.metrics.nlp import LanguageCrossEntropy, MaskedAccuracy
+from composer.models import HuggingFaceModel
+from composer.trainer import Trainer
+from composer.utils import dist
+from tests.common import device, world_size
+
+
+@pytest.mark.daily
+@pytest.mark.remote
+@device('cpu', 'gpu')
+@world_size(1, 2)
+@pytest.mark.parametrize('num_workers', [0, 1, 2])
+@pytest.mark.parametrize('dataset,dataset_args,seed',
+                         [('c4', {
+                             'remote': 's3://mosaicml-internal-dataset-c4/mds/2/',
+                             'tokenizer_name': 'bert-base-uncased',
+                             'max_seq_len': 256,
+                             'group_method': 'truncate'
+                         }, 1),
+                          ('pile', {
+                              'remote': 's3://mosaicml-internal-dataset-the-pile/mds/2/',
+                              'tokenizer_name': 'bert-base-uncased',
+                              'max_seq_len': 256,
+                              'group_method': 'truncate'
+                          }, 2), ('enwiki', {
+                              'remote': 's3://mosaicml-internal-dataset-enwiki-20200101/mds/2b/'
+                          }, 3)])
+def test_streaming_datasets(num_workers, dataset, dataset_args, seed, tiny_bert_tokenizer, tiny_bert_model, world_size,
+                            device, tmp_path):
+    # Need to initialize dist before we get to streaming, because streaming always uses NCCL
+    if not dist.is_initialized():
+        dist.initialize_dist(device=device)
+
+    streaming = pytest.importorskip('streaming')
+    transformers = pytest.importorskip('transformers')
+    name_to_cls = {
+        'c4': streaming.text.c4.StreamingC4,
+        'pile': streaming.text.pile.StreamingPile,
+        'enwiki': streaming.text.enwiki.StreamingEnWiki
+    }
+
+    full_seed = seed + (num_workers + 1) * 10 + (world_size + 1) * 100 + (1 if device == 'cpu' else 2) * 1000
+    # This seed setting is necessary to prevent a shared memory collision due to a streaming bug
```

Comment:
This exact seed???

Reply:
Resolving for now, but please follow up @dakinggg

Reply:
Not this exact seed. But the seeds needed to be unique across all parametrizations of this test. I think this has been fixed in the latest streaming, and I made a JIRA to clean up this test with the latest version of streaming